tests/integration/security/rbac/util/rbac_util.go

```diff
@@ -65,9 +65,15 @@ func (tc TestCase) CheckRBACRequest() error {
 			return getError(req, "allow with code 200", fmt.Sprintf("error: %v", err))
 		}
 	} else {
-		if req.Options.PortName == "tcp" {
-			if err == nil || !strings.Contains(err.Error(), "EOF") {
-				return getError(req, "deny with EOF error", fmt.Sprintf("error: %v", err))
+		if req.Options.PortName == "tcp" || req.Options.PortName == "grpc" {
+			expectedErrMsg := "EOF" // TCP deny message.
+			if req.Options.PortName == "grpc" {
+				expectedErrMsg = "rpc error: code = PermissionDenied desc = RBAC: access denied"
```

Comment:
Do you need to "return getError" for this condition? Where is expectedErrMsg used?

Reply:
It's used right below https://github.com/istio/istio/pull/14371/commits/41bd4196ae3efeb4b6f5e04feb6df1aba7656651#diff-61c4b1adf3ed4ea8a393487c211f7d8eL78

Reply:
I missed it. Thanks for explanation.